packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -119,7 +119,7 @@ const isBlockActive = (editor, matchNode) => {
   return match.length > 0;
 };
 
-const toggleBlock = (editor, value) => {
+const setBlock = (editor, value) => {
```

Comment:
If these functions do need to be outside of the component I think they should be just outside the component that is using them not at the top of the file. Which makes me think maybe we could organize this file better by adding these style comments:

https://github.com/strapi/strapi/blob/24fa848ba3d5bddad1f8eb56b79cfb41053ad350/packages/core/helper-plugin/src/components/Table.tsx#L140-L142

It makes the file easier to read and allows to organize code outside components that pertain to a specific component. For example these functions or image constants and pick utility

Reply:
@markkaylor I don't believe `addBlockToEditor` would be accurate, because I think this selector doesn't create a new block but changes the type of the current selection instead. I get your point about it sounding like useState though. Maybe `convertBlock` could work?

Reply:
Another thing to keep in mind is that we'll soon build the slash command which will require reusing this select outside of the toolbar. So it will likely need to be moved somewhere higher in the file tree. So moving toggleBlocks and isBlockActive inside the component will make that easier

Reply:
I'm cool with that but I think it won't always be converting either. Sometimes it is and sometimes it is adding a new block right? Do you not see any value in specifying where the action is happening. I guess with a JSDoc description it's probably not necessary.

Reply:
Yes, you are right @markkaylor I changed the name because toggleBlock was wrong, the function handles just the change of the option selected in the dropdown, maybe we can call it `selectBlock`

Reply:
>Sometimes it is and sometimes it is adding a new block right?

@markkaylor i don't think it ever adds a new block 🤔 

when you are on a new line it converts the empty paragraph node (that was created by pressing enter) into what you select

Reply:
selectBlock works for me 👍 @simotae14 

@remidej I forgot I added that on my PR 🤦‍♂️ 
https://github.com/strapi/strapi/pull/18193

Let's discuss it over there. Here's the case: If the editor has never been focused and you select a block nothing will happen, so I added a condition but maybe it's not the XP we want.

Reply:
> If these functions do need to be outside of the component I think they should be just outside the component that is using them not at the top of the file. Which makes me think maybe we could organize this file better by adding these style comments:
> 
> https://github.com/strapi/strapi/blob/24fa848ba3d5bddad1f8eb56b79cfb41053ad350/packages/core/helper-plugin/src/components/Table.tsx#L140-L142
> 
> It makes the file easier to read and allows us to organize code outside components that pertain to a specific component. For example these functions would go under a comment for the BlocksDropdown component, or the image constants and pick utility would go under the comment for the ImageDialog component, etc...

For me this file is very messy and I always get lost, I prefer to have separate components instead and just have simple toolbar component(single responsiblity). Its easier to unit test as well independently if required. 
Nobody noticed `ModifierButton` component is no more used in the BlockEditor and its still lying there.

Reply:
I agree the file is difficult to navigate. I have mixed feelings about the colocation but not a strong opinion. I think it can also be organized nicely as well and reduce the number of files you need open. That being said your point about the dead code is valid.

I believe it was setup this way to respect the [FE guidelines](https://contributor.strapi.io/guides/fe-coding-guidelines#application-structure) to keep code together and only move when needed elsewhere:

> When components/hooks etc. are required higher in the tree, we can subsequently move them higher up as necessary. But we strive for colocation.

It's my understanding that any components we move out of this file _will_ require a unite test.

@joshuaellis @remidej I would be interested in your opinions here on how we can organize and make this easier to work with.

Reply:
The issue with having separate files is you create very shallow modules and a wider API surface overall. Realistically we should be creating deep modules with a clear API surface.

I think there are a few things we can do to make this better, the naming of all these components is somewhat inconsistent while it's clear, we could create more harmony, adding comment blocks like @markkaylor suggested is another way to create some visual distinction. There's a few other ideas like do we need a separate function of `isListActive` or can we just inline the calculation so there's no "where is this function written".

I'm happy to do a group session Monday or Tuesday all together and we can work on this together and discuss what we're doing and why :) 

An example of what it could end up looking like would be this: https://github.com/strapi/design-system/blob/main/packages/primitives/src/components/Combobox/Combobox.tsx